tree.go

```diff
@@ -269,7 +269,18 @@ walk:
 // Returns -1 as index, if no wildcard was found.
 func findWildcard(path string) (wildcard string, i int, valid bool) {
 	// Find start
+	escapeColon := false
 	for start, c := range []byte(path) {
+		if escapeColon {
+			if c == ':' {
+				escapeColon = false
```

Comment:
What about a case of `\\foo:`? `escapeColon` would stay true after escape character until colon is reached. It's probably not expected behaviour. It should try to escape next character by the rules. This case is also not covered with test.

Reply:
test case added 
https://github.com/gin-gonic/gin/blob/e0d9950e7a69d967ac80f0395df2f784e3c4cdb6/gin_integration_test.go#L592